samples/subsys/video/capture_to_lvgl/src/main.c

```diff
@@ -0,0 +1,213 @@
+		       (char)(fcap->pixelformat >> 16),
+		       (char)(fcap->pixelformat >> 24),
+		       fcap->width_min, fcap->width_max, fcap->width_step,
+		       fcap->height_min, fcap->height_max, fcap->height_step);
+		i++;
+	}
+
+	/* Get default/native format */
+	if (video_get_format(video_dev, VIDEO_EP_OUT, &fmt)) {
+		LOG_ERR("Unable to retrieve video format");
+		return 0;
+	}
+
+	printk("- Default format: %c%c%c%c %ux%u\n", (char)fmt.pixelformat,
+	       (char)(fmt.pixelformat >> 8),
+	       (char)(fmt.pixelformat >> 16),
+	       (char)(fmt.pixelformat >> 24),
+	       fmt.width, fmt.height);
+
+	/* Set format */
+	fmt.width = CONFIG_VIDEO_WIDTH;
+	fmt.height = CONFIG_VIDEO_HEIGHT;
+	fmt.pitch = fmt.width * 2;
+
+	if (video_set_format(video_dev, VIDEO_EP_OUT, &fmt)) {
+		LOG_ERR("Unable to set up video format");
+		return 0;
+	}
+
+	/* Get format */
+	if (video_get_format(video_dev, VIDEO_EP_OUT, &fmt)) {
+		LOG_ERR("Unable to retrieve video format");
+		return 0;
+	}
+
+	printk("- New format: %c%c%c%c %ux%u %u\n", (char)fmt.pixelformat,
+		(char)(fmt.pixelformat >> 8),
+		(char)(fmt.pixelformat >> 16),
+		(char)(fmt.pixelformat >> 24),
+		fmt.width, fmt.height, fmt.pitch);
+
+	/* Size to allocate for each buffer */
+	bsize = fmt.pitch * fmt.height;
+
+	/* Alloc video buffers and enqueue for capture */
+	for (i = 0; i < ARRAY_SIZE(buffers); i++) {
+		buffers[i] = video_buffer_alloc(bsize);
+		if (buffers[i] == NULL) {
+			LOG_ERR("Unable to alloc video buffer");
+			return 0;
+		}
+
+		video_enqueue(video_dev, VIDEO_EP_OUT, buffers[i]);
+	}
+
+	/* Start video capture */
+	if (video_stream_start(video_dev)) {
+		LOG_ERR("Unable to start capture (interface)");
+		return 0;
+	}
+
+	display_splash_screen();
+
+	display_blanking_off(display_dev);
+
+	const lv_img_dsc_t video_img = {
+		.header.always_zero = 0,
+		.header.w = CONFIG_VIDEO_WIDTH,
+		.header.h = CONFIG_VIDEO_HEIGHT,
+		.data_size = CONFIG_VIDEO_WIDTH * CONFIG_VIDEO_HEIGHT * sizeof(lv_color_t),
+		.header.cf = LV_IMG_CF_TRUE_COLOR,
+		.data = (const uint8_t *) buffers[0]->buffer,
+	};
+
+	lv_obj_t *screen = lv_img_create(lv_scr_act());
+
+	printk("Capture started\n");
+
+	/* Initialize FPS counter and timer */
+	uint32_t fps = 0;
+	uint32_t frame_counter = 0;
+	uint32_t last_time = k_uptime_get_32();
+
+	/* Grab video frames */
+	while (1) {
+		int err;
+
+		err = video_dequeue(video_dev, VIDEO_EP_OUT, &vbuf, K_FOREVER);
+		if (err) {
+			LOG_ERR("Unable to dequeue video buf");
+			return 0;
+		}
+
+		lv_img_set_src(screen, &video_img);
+		lv_obj_align(screen, LV_ALIGN_BOTTOM_LEFT, 0, 0);
+
+		frame_counter++;
+
+		/* Calculate FPS every second */
+		if (k_uptime_get_32() - last_time > 1000) {
+			fps = frame_counter;
+
+			/* Reset frame counter and timer */
+			frame_counter = 0;
+			last_time = k_uptime_get_32();
+		}
+
+		printk("\rFrame: %u, FPS: %u, Size: %u bytes, Timestamp: %u ms",
+			frame++, fps, vbuf->bytesused, vbuf->timestamp);
```

Comment:
related to my comment on showing "more" lvgl stuff -- I would totally display the framerate on the screen... in fact, you can probably get rid of your custom code altogether and simply enable `CONF_LV_USE_PERF_MONITOR` :)

Reply:
Do you think I should have removed this print message and enabled CONF_LV_USE_PERF_MONITOR?

![photo_5028563151020469497_y](https://github.com/user-attachments/assets/f79b49e3-fe99-480b-9522-ec9145e33b00)

Reply:
I think it makes a lot of sense to just rely on upstream LVGL functionality whenever possible, no? :)
Cool to see it in action!

Reply:
Removed the previous log lines and enabled the CONF_LV_USE_PERF_MONITOR.